pkg/trace/api/api.go

```diff
@@ -390,6 +402,38 @@ func (r *HTTPReceiver) rateLimited(n int64) bool {
 	return !r.RateLimiter.Permits(n)
 }
 
+// StatsProcessor implementations are able to process incoming client stats.
+type StatsProcessor interface {
+	// ProcessStats takes a stats payload and consumes it. It is considered to be originating
+	// from the given lang.
+	ProcessStats(p pb.ClientStatsPayload, lang string)
+}
+
+// handleStats handles incoming stats payloads.
+func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
+	defer timing.Since("datadog.trace_agent.receiver.stats_process_ms", time.Now())
+
+	ts := r.tagStats(v05, req)
+	rd := NewLimitedReader(req.Body, r.conf.MaxRequestBytes)
+	metrics.Count("datadog.trace_agent.receiver.stats_payload", 1, ts.AsTags(), 1)
```

Comment:
Is there a reason why stats for handleStats are reported differently than in handleTraces?

Reply:
Your questions is a bit ambiguous but I'll assume you refer to the atomic values. We use atomic values in hot paths which I think is not the case for either endpoint, especially not for this one.

Reply:
Yes that's what I was referring to, thank you for the explanation. That's maybe work for another PR but I think we should make it consistent if there is no reason to have two different ways of reporting similar metrics.